### PR TITLE
selecting best utterance from alternatives

### PR DESCRIPTION
**Pull request by bigdatabaracus**
selecting best utterance from alternatives

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -132,6 +134,31 @@ const findProducts = (utteranceData, languageCode) => {
...
+
+  // you can tune alpha between 0.0 - 1.0 to control how much
+  // the product score influences which utterance alternative is selected
+  const alpha = 0.1;
+  const bestResult =
+    searchResults.length > 0
+      ? searchResults
+          .map((result, i) => ({ result: result, score: Math.log(confidences[i]) + alpha * Math.log(scores[i]) }))
+          .reduce((best, current, i) => (i === 0 ? current : current.score > best.score ? current : best))
+      : [];
```
Comment (langma) [inline index.js]: Breaks in return if `searchResults` is empty.

File: index.js [hunk 2]
```diff
@@ -144,7 +171,7 @@ const processData = (websocket, languageCode) => {
     } else if (data.finished !== undefined) {
       websocket.send(JSON.stringify({ event: "stopped", data: data.finished }));
     } else {
-      const productSegments = findProducts(data.utterance, languageCode);
+      const productSegments = parseUtterance(data.utterance, languageCode);
```
Comment (langma) [inline index.js]: I think a function named `parseUtterance` should return an utterance, not products.

File: index.js (hunk 1 above)
Comment (bigdatabaracus) [reply index.js]: well spotted. Fixed.

**Comment by bigdatabaracus** [comment]
Thank you for the comments @langma. Check if now better.

**Pull request merged**